junit5-vanilla-modules/build.jsh

```diff
@@ -0,0 +1,96 @@
+args.addAllFiles("src/test", ".java")
+run("javac", args.toArray())
+
+args = new Arguments()
+args.add("--module-path").addPath("bin/test", "bin/main", "lib")
+args.add("--add-modules").add("ALL-MODULE-PATH,ALL-DEFAULT")
+args.add("--module").add("org.junit.platform.console")
+args.add("--scan-modules")
+args.add("--reports-dir").add("bin/test-results/junit-platform")
+exe("java", args.toArray())
+
+//
+// [test-classpath] compile and run tests on the classpath
+//
+args = new Arguments()
+args.add("-d").add("bin/test-classpath")
+args.add("--class-path").addPath("bin/main/com.example.application", "bin/main/com.example.tool", "lib/junit-jupiter-api-5.1.0.jar", "lib/apiguardian-api-1.0.0.jar", "lib/opentest4j-1.0.0.jar", "lib/junit-4.12.jar")
+args.addAllFiles("src/test", path -> { String name = path.getFileName().toString(); return name.endsWith(".java") && !name.contains("module-info"); } )
+run("javac", args.toArray())
+
+args = new Arguments()
+args.add("--class-path").addPath("bin/test-classpath", "bin/main/com.example.application", "bin/main/com.example.tool", "lib/*")
+args.add("org.junit.platform.console.ConsoleLauncher")
+args.add("--scan-class-path")
+args.add("--reports-dir").add("bin/test-classpath-results/junit-platform")
+exe("java", args.toArray())
+
+
+//
+// [test-patched] compile and run tests on the classpath
+//
+args = new Arguments()
+args.add("-d").add("bin/test-patched/com.example.application")
+args.add("--class-path").addPath("bin/main/com.example.application", "bin/main/com.example.tool", "lib/junit-jupiter-api-5.1.0.jar", "lib/apiguardian-api-1.0.0.jar", "lib/opentest4j-1.0.0.jar", "lib/junit-4.12.jar")
+args.addAllFiles("src/test/com.example.application", path -> { String name = path.getFileName().toString(); return name.endsWith(".java") && !name.contains("module-info"); } )
+run("javac", args.toArray())
+args = new Arguments()
+args.add("-d").add("bin/test-patched/com.example.tool")
+args.add("--class-path").addPath("bin/main/com.example.application", "bin/main/com.example.tool", "lib/junit-jupiter-api-5.1.0.jar", "lib/apiguardian-api-1.0.0.jar", "lib/opentest4j-1.0.0.jar", "lib/junit-4.12.jar")
+args.addAllFiles("src/test/com.example.tool", path -> { String name = path.getFileName().toString(); return name.endsWith(".java") && !name.contains("module-info"); } )
+run("javac", args.toArray())
+
+args = new Arguments()
+args.add("--module-path").addPath("bin/main", "lib", "bin/test/black.box") // re-use real "black.box" module here
+args.add("--add-modules").add("ALL-MODULE-PATH,ALL-DEFAULT")
+args.add("--patch-module").add("com.example.application=bin/test-patched/com.example.application")
+args.add("--patch-module").add("com.example.tool=bin/test-patched/com.example.tool")
+args.add("--add-opens").add("com.example.application/com.example.application=org.junit.platform.commons")
+args.add("--add-opens").add("com.example.tool/com.example.tool=org.junit.platform.commons")
```

Comment:
@nicolaiparlog Why do I have to explicitly name `org.junit.platform.commons` in the `--add-opens` options? Should `ALL-UNNAMED` according to https://blog.codefx.org/java/five-command-line-options-to-hack-the-java-9-module-system/#Reflectively-Accessing-Internal-APIs-With--add-opens be sufficient?